src/lua/minifio.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+struct lua_State;
```

Comment:
I need the type name defined to define a pointer argument. It isn't worth to pull the definition of the type and all the content of the `lua.h` header into this header file. It is usual for header files to define types this way to keep things more clean and to speedup compilation. IOW, I have no goal to avoid this forward declaration, but I think it is a good practice to avoid `#include <lua.h>` here.